### PR TITLE
fix(VTextField): properly detect activeElement from shadowRoot

### PR DESCRIPTION
**Pull request by matthieusieben**
## Description

When an `input` element is inside a shadowDom, and is active (focused), the `document.activeElement` will not reference the input element but rather the shadowRoot. In order to properly detect what the `activeElement` is, we must use the proper root node for that element, as explained on [MDN](https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/activeElement).

## Motivation and Context

This is part of the solution for:
- #7622

And an alternative implementation to:
- #12141
- #12142

## How Has This Been Tested?

unit

## Markup:

<details>

```vue
// Paste your FULL Playground.vue here
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by matthieusieben** [comment]
Tests are failing because `el.getRootNode()` returns `el` in the test suite (not sure where the jsdom is instantiated).

There are two options:
```js
    getActiveElement () {
      if (!this.$el.getRootNode) return document // IE11

      const rootNode = this.$el.getRootNode() as ShadowRoot | HTMLDocument
      if (rootNode === this.$el) return document.activeElement // jest

      return rootNode.activeElement
    },
```

or ignore this line in coverability

**Comment by matthieusieben** [comment]
Sorry about the noise with the 3 PRs. I am just not sure how you want to go about this, so I proposed 3 solutions

**Comment by johnleider** [comment]
The separate PR's are preferred, ty.

**Comment by matthieusieben** [comment]
What do you mean? @johnleider

**Comment by KaelWD** [comment]
- What's the difference between #12141 and #12142
 - Is all that IE stuff necessary? #12141 is much simpler and looks like it should work just as well.

**Comment by johnleider** [comment]
This Pull Request is being closed due to inactivity.

If you have any additional questions, please reach out to us in our [Discord community](https://community.vuetifyjs.com).

**Comment by matthieusieben** [comment]
I proposed two solutions to one of the issues related to #7622:
- a basic solution (#12141 / #12142)
- a safer solution (#12134)

- #12134: detects if the node is actually attached in the dom before trying to focus it. Doing this allows tests to work seamlessly wether the node is attached to the DOM or not, and only actually attaches the node to the DOM in focus related tests.
- #12141 / #12142: assumes a node is always mounted to the dom
  - The difference between the two is only regarding the tests:
    - #12141: this one will *always* attach elements to the dom in all input elements related tests. It makes more sense to always attach input elements to the DOM to properly test them as the behavior might vary. The drawback of this is that it "enforses" that all input related tests should always attach the element to the dom to be consistent.
    - #12142: this one attaches elements to the dom only in tests that would fail otherwise (= tests related to the focus of the input). If is a less clean solution but has a lighter footprint.

TL;DR
- #12134: safer solution & lighter test footprint
- #12142: basic solution & lighter test footprint
- #12141: basic solution & higher test footprint


IMO this one should be kept as it has both a safer implementation and lighter footprint.

**Comment by matthieusieben** [comment]
@johnleider could you consider re-opening this

**Comment by matthieusieben** [comment]
Created a new PR in https://github.com/vuetifyjs/vuetify/pull/13053 that includes more details
